presto-main/src/main/java/com/facebook/presto/execution/TaskExecutor.java

```diff
@@ -87,6 +88,30 @@
 
     private final Ticker ticker;
 
+    private static class SplitProfile
+        implements Comparable<SplitProfile>
+    {
+        private final long startTime;
+        private final long splitId;
+
+        private SplitProfile(long startTime, long splitId)
+        {
+            this.startTime = startTime;
+            this.splitId = splitId;
+        }
+
+        @Override
+        public int compareTo(SplitProfile o)
+        {
+            if (startTime == o.startTime) {
```

Comment:
You can do it like this:
```java
return ComparisonChain.start()
        .compare(startTime, o.startTime)
        .compare(splitId, o.splitId)
        .result();
```
Note that if you needed a `Comparator`, Java 8 has a similarly nice way to do this (though using `ComparisonChain` seems nicer for implementing `Comparable`):
```java
Comparator.comparingLong(SplitProfile::getStartTime)
        .thenComparingLong(SplitProfile::getSplitId);
```